src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java

```diff
@@ -286,55 +286,60 @@ JsonWebKey createJwkFromSettings(Settings settings, Settings jwtSettings) throws
                 );
             }
 
-            JsonWebKey jwk = new JsonWebKey();
+            //TODO nimbus library doesn't accept generic key creation. Decide between RSAKey, OCTET, ECKey and create key from raw settings
```

Comment:
I'd recommend to keep with Octet for the moment

Reply:
Currently there is support for for HMAC and RSA. This part of code was taking raw settings and putting it into generic cxf JWK. Nimbus library doesn't allow that generic implementation, so I'll add in-code check for RSA/HMAC algo and build object using proper constuctor.